### ISSUE TITLE
Unable to read env variables set in config file

### ISSUE THREAD

**Issue by RokubungiGendo** [ballerina-platform/ballerina-lang#26913]
**Description:**
Unable to read env variables set in config file.. sorry be nice to me is my first issues

**Steps to reproduce:**

**Src file**
import ballerina/io;
import ballerina/config;

public function main() {
string confenv = config:getAsString("testenv.envvar");
string confval = config:getAsString("testenv.var");
io:println("Configuration variable: "+ confval);
io:println("Configuration variable from env: "+ confenv);
}

**Config file**
[testenv]
envvar="@env{TEST_ENV}"
var="TESTVAR"

**Set env variable in PS**
$env:TEST_ENV="foo"

**Run ballerina**
ballerina run testenv.bal

**Result**
Compiling source
testenv.bal
Running executables

Configuration variable: TESTVAR
Configuration variable from env: @env{TEST_ENV}

**Affected Versions:**
1.2.9

**OS, DB, other environment details and versions:**
Windows 10 Pro 20H2 build 19042.630 - PowerShell 5.1.19041.610 - Ballerina 1.2.9

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by pubudu91** [comment]
IIRC, we don't support the following anymore: `envvar="@env{TEST_ENV}"`

What you can do is, instead of explicitly setting a value in the config file just to read an env var, directly read the env var from the config API. For example, in this case it'd be,
```ballerina
string confenv = config:getAsString("TEST_ENV");
```

However, if you do want to take advantage of the config hierarchy, the env var name has to match the config key in the file. For example,
```Config file
[testenv]
envvar="bar"
var="TESTVAR"

Set env variable in PS
$env:testenv_envvar="foo"
```
Note the format for the env var name. In the code, we'd refer to the config key as `testenv.envvar`. But since `.` is not allowed in env var names, what we do when we look up env vars is that we replace `.` with `_`. Hence, the env var name `testenv_envvar`.

Hope this clears things up for you :)

**Comment by RokubungiGendo** [comment]
thanks, the proposed solution works great
I found the documentation for this function here:
[https://ballerina.io/learn/api-docs/ballerina/config/index.html](url)
was I wrong or is this page out of date?

**Comment by pubudu91** [comment]
Oh wow. Yeah, that page looks like it needs to be updated. We took that feature off a long time back. Thanks for pointing it out. We'll fix it. 

Additionally, if you are only working with env vars, you can also just simply use the `system:getEnv()` function: https://ballerina.io/learn/api-docs/ballerina/system/functions.html#getEnv

**Comment by pubudu91** [comment]
Closing since this was fixed with #27155
